.github/workflows/main.yaml

```diff
@@ -1,4 +1,4 @@
-name: Lint CI
+name: Main CI
```

Comment:
Since you're changing this, you may want to run [zizmor](https://github.com/zizmorcore/zizmor) on it and add those changes too--or maybe as a separate PR.

It will at minimum suggest setting the `permissions` object to something (as the defaults are more that what is needed here).